### PR TITLE
initial spike for a search feature

### PR DESCRIPTION
**Pull request by joelhooks**
Rudimentary search, but it works well!

How can we integrate it?

Search Total TypeScript  Total TypeScript 2023-05-02 at 14 55 38@2x

spike

### CODE REVIEW

File: apps/total-typescript/src/trpc/routers/search.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        `  *[_type in ["article", "tip", "module", "exercise", "explainer"]]
+    | score(
+      title match $searchQuery 
+      || _type match "module"
+      || description match $searchQuery
+      || pt::text(body) match $searchQuery
+      || boost(pt::text(body) match $searchQuery + "*", 0.5)
+      || boost(pt::text(transcript) match $searchQuery, 0.5)
+
+    )
```
Comment (joelhooks) [inline apps/total-typescript/src/trpc/routers/search.ts:25]: this gives you a quick rank and seems to work very well.

File: apps/total-typescript/src/trpc/routers/search.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      
+    } | {
+      ...,
+      "module": *[_type in ['module'] && references(^.section._id)][0] {
+        _id,
+        slug,
+        title,
+        moduleType
+      }
+    }
```
Comment (joelhooks) [inline apps/total-typescript/src/trpc/routers/search.ts:49]: we can have access to the section and module for link building

File: apps/total-typescript/src/trpc/routers/search.ts (hunk 1 above)
Comment (vojtaholik) [reply apps/total-typescript/src/trpc/routers/search.ts:25]: pretty cool!

**Review by vojtaholik — APPROVED** [review]
I wanna try making this available site wide via [kbar](https://github.com/timc1/kbar); top nav would get a search bar with keyboard shortcut hint that would trigger it open.

**Pull request merged**